File: bugsnag-android-core/src/test/java/com/bugsnag/android/NotifierTest.java
package com.bugsnag.android;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

import org.junit.Before;
import org.junit.Test;

public class NotifierTest {

    private Notifier notifier;

    @Before
    public void setUp() throws Exception {
        notifier = new Notifier();
    }

    @Test
    public void testName() {
        assertEquals("Android Bugsnag Notifier", notifier.getName());
        String expected = "CrossPlatformFramework";
        notifier.setName(expected);
        assertEquals(expected, notifier.getName());
    }

    @Test
    public void testVersion() {
        assertNotNull(notifier.getVersion());
        String expected = "1.2.3";
        notifier.setVersion(expected);
        assertEquals(expected, notifier.getVersion());
    }

    @Test
    public void testUrl() {
        assertEquals("https://bugsnag.com", notifier.getURL());
        String expected = "http://example.com";
        notifier.setURL(expected);
        assertEquals(expected, notifier.getURL());
    }
}

File: bugsnag-android-core/src/test/java/com/bugsnag/android/NotifierTest.kt
package com.bugsnag.android

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull

import org.junit.Before
import org.junit.Test

class NotifierTest {

    private val notifier = Notifier

    @Test
    fun testName() {
        assertEquals("Android Bugsnag Notifier", notifier.name)
        val expected = "CrossPlatformFramework"
        notifier.name = expected
        assertEquals(expected, notifier.name)
    }

    @Test
    fun testVersion() {
        assertNotNull(notifier.version)
        val expected = "1.2.3"
        notifier.version = expected
        assertEquals(expected, notifier.version)
    }

    @Test
    fun testUrl() {
        assertEquals("https://bugsnag.com", notifier.url)
        val expected = "http://example.com"
        notifier.url = expected
        assertEquals(expected, notifier.url)
    }
}
